_src/_posts/2018-09-07-writing-custom-presto-functions.markdown

```diff
@@ -0,0 +1,108 @@
+---
+layout: post
+title:  Writing custom PrestoDB functions
+date:   2018-09-07
+tag:    technology
+categories:
+    - infrastructure
+    - technology
+keywords: "Presto, PrestoDB, UDF, aggregation"
+author: dantepawlow
+---
+
+# Writing custom PrestoDB functions
+
+<!--excerpt.start-->
+
+Here at Jampp we process and analyse large amounts of data.
+One of the tools we employ to do so is [PrestoDB](https://prestodb.io/), which is a "Distributed SQL Query Engine for Big Data".
+Presto comes with many [native functions](https://prestodb.io/docs/current/functions.html), which are usually enough for most use cases. Nevertheless, sometimes you need to implement your own function for a very specific use.
+
+Enter the [User Defined Functions](https://prestodb.io/docs/current/develop/functions.html) (UDFs, for short).
+Writing one is not as straightforward as it first appears, mainly because the information to do so is very scattered around the web (and across many Presto versions).
+
+In this blogpost, we present our *JSON_Sum* function, how did we write it, and some of the lessons we learned along the way.
+
+<!--excerpt.end-->
+
+## Function types
+
+Presto has two main types of functions: scalar and aggregation[^1].
+Scalar functions behave more like traditional functions, in the sense that their output is immediate from the values of their parameters.
```

Comment:
Thinks some alternative for immediate.  I don't think it's really clear what you are trying to say.

Reply:
"Scalar functions behave more like traditional functions, in the sense that their output is intuitively derived from their parameters."?